app/controllers/invitations_controller.rb

```diff
@@ -9,7 +9,7 @@ def create
     params[:invitation].merge!(inviter_id: current_user.id)
     @invitation = Invitation.new(invitation_params)
     if @invitation.save
-      UserMailer.delay.send_friend_invitation(@invitation)
+      UserMailer.send_friend_invitation(@invitation.id).deliver
```

Comment:
You should send the invitation email via background jobs. This was not the root of your issues before.